src/declaration.c

```diff
@@ -1653,6 +1653,7 @@ void VarDeclaration::setFieldOffset(AggregateDeclaration *ad, unsigned *poffset,
             {
                 const char *s = (t->ty == Tsarray) ? "static array of " : "";
                 ad->error("cannot have field %s with %ssame struct type", toChars(), s);
+                return;
```

Comment:
I don't like adding too verbose comment in error recovery path. If someone will accidentally remove the return, the added test case will fail. It's enough.